src/phpDocumentor/ConfigurationFactory.php

```diff
@@ -0,0 +1,354 @@
+<?php
+
+namespace phpDocumentor;
+
+final class ConfigurationFactory
+{
+    /**
+     * @var Uri
+     */
+    private $uri;
+
+    /**
+     * @var string
+     */
+    private $schemaPath;
+
+    /**
+     * @param Uri    $uri
+     * @param string $schemaPath
+     */
+    public function __construct(Uri $uri, $schemaPath = '../../data/xsd/phpdoc.xsd')
+    {
+        $this->validate($uri);
+
+        $this->uri        = $uri;
+        $this->schemaPath = $schemaPath;
+    }
+
+    /**
+     * Replace the location of the configuration file.
+     *
+     * @param Uri $uri
+     *
+     * @return ConfigurationFactory
+     */
+    public function replaceLocation(Uri $uri)
+    {
+        return static::__construct($uri);
```

Comment:
The second parameter ($schemaPath) of __construct is a property with a default value; you omit the second parameter here so it will assume that default value. Subsequently that default value is set on the property and will overwrite the $schemaPath that you have set when you created the object